### PR TITLE
chore(deps): angular18 upgrade

### PR DESCRIPTION
**Pull request by GuillaumeNury**
## Description

Angular and friends update.

-----

This PR has some noise as I had to bump eslint/prettier deps.

I depend on an early build of angular-eslint, but : 
1. It is a dev dependency
2. It should not change in a breaking way anymore ([quote from maintainer](https://github.com/angular-eslint/angular-eslint/pull/1830#issuecomment-2135134938))

-----

### CODE REVIEW

File: packages/ng/popup-employee/service/user-popover.store.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export class LuUserPopoverStore implements ILuUserPopoverStore {
 
 	public get(id: number): Observable<LuUserPopover> {
 		if (this.#cache.has(id)) {
-			return this.#cache.get(id)!;
+			return this.#cache.get(id);
```
Comment (shajz) [inline packages/ng/popup-employee/service/user-popover.store.ts:19]: woah

File: setup-jest.ts [hunk 2]
```diff
@@ -1,11 +1,14 @@
 import '@testing-library/jest-dom';
 import { toHaveNoViolations } from 'jest-axe';
 import 'jest-preset-angular/setup-jest';
+import { TextDecoder, TextEncoder } from 'util';
 
 expect.extend(toHaveNoViolations);
 
 Element.prototype.scrollIntoView = () => {};
 
+Object.assign(global, { TextDecoder, TextEncoder });
```
Comment (shajz) [inline setup-jest.ts:10]: why do we need this? Maybe add a comment here

File: packages/ng/popup-employee/service/user-popover.store.ts (hunk 1 above)
Comment (GuillaumeNury) [reply packages/ng/popup-employee/service/user-popover.store.ts:19]: it's a great news, no more not null assertion when `get` is guarded inside a `has` 😄

**Review by shajz — APPROVED** [review]

File: setup-jest.ts (hunk 2 above)
Comment (GuillaumeNury) [reply setup-jest.ts:12]: done

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2823/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/17706/ui-diff)

**Pull request merged**